src/crowdin_gui.cpp

```diff
@@ -602,20 +604,33 @@ void CrowdinSyncFile(wxWindow *parent, std::shared_ptr<Catalog> catalog,
     // TODO: nicer API for this.
     // This must be done right after entering the modal loop (on non-OSX)
     dlg->CallAfter([=]{
+        const auto& header = catalog->Header();
+        auto crowdin_lang = header.HasHeader("X-Crowdin-Language")
+                            ? Language::TryParse(header.GetHeader("X-Crowdin-Language").ToStdWstring())
+                            : catalog->GetLanguage();
+        CrowdinUpdateCatalogIDsFromHeaders(catalog).then([=]() {
         CrowdinClient::Get().UploadFile(
-                str::to_utf8(crowdin_prj), str::to_wstring(crowdin_file), crowdin_lang,
+                catalog->GetCrowdinProjectId(),
+                crowdin_lang,
+                catalog->GetCrowdinFileId(),
+                std::string(wxFileName(catalog->GetFileName()).GetExt().utf8_str()),
                 catalog->SaveToBuffer()
             )
             .then([=]{
+                wxFileName filename = catalog->GetFileName();
                 auto tmpdir = std::make_shared<TempDirectory>();
-                auto outfile = tmpdir->CreateFileName("crowdin.po");
+                auto outfile = tmpdir->CreateFileName("crowdin." + filename.GetExt());
 
                 dispatch::on_main([=]{
                     dlg->Activity->Start(_(L"Downloading latest translations…"));
                 });
-
+                if(filename.GetExt().CmpNoCase("po"))// if not PO
+                    filename.SetFullName(filename.GetName());// set original filename extension
```

Comment:
How does this line "set original filename extension" and what is "original" extension?

Please use explicit '!= 0' tests in cmp functions, so that casual reader doesn't read them as returning boolean, or use wxString::IsSameAs() instead.

Reply:
Addressed 69df2aa7dd6996e4505240e263d0f1dac167b489 , including more clear comment now (though please feel free to `Unresolve` if still unclear)